build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -302,84 +287,34 @@ private void setDistributionType(ElasticsearchDistribution distribution, TestDis
         }
     }
 
-    // package protected so only TestClustersAware can access
```

Comment:
Since we never have node specific plugin or module requirements I moved this one level up to ElasticsearchCluster. That reduces the API surface and makes this handling more efficient as we only do this whole configuration setup once per cluster.